### PR TITLE
feat: add python templates

### PR DESCRIPTION
**Pull request by lance**
This commit adds support for Python by including basic function/test templates for events and http. The event function seems to highlight that creating a simple invocation framework (as opposed to the server.py in buildpacks repo) to help users navigate the event universe might be a good idea.

Signed-off-by: Lance Ball <lball@redhat.com>

### CODE REVIEW

**Comment by lance** [comment]
I should add runtime documentation to this before we land it.

**Review by lkingland — APPROVED** [review]
Looks good to me.  So nice to see we now support Python too; this is an important runtime.

**Review by zroubalik — APPROVED** [review]
Great job!

**Comment by matejvasek** [comment]
Looks good to me.

**Review by matejvasek — APPROVED** [review]

File: templates/python/events/func.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+from cloudevents.http import CloudEvent, to_binary
+
+# ---------------------------------------
+# Function template
+# context is a dictionary with the Flask request object and 
+# and a cloud_event key for incoming events
+# ---------------------------------------
+def main(context):
```
Comment (matejvasek) [inline templates/python/events/func.py:8]: Just idea:
What about using type hints?
```python3
from werkzeug import Request
from typing import TypedDict


class Context(TypedDict):
    request: Request


def main(context: Context):
    """
    Doc here
    :param context: 
    :return: 
    """
    return {"message": "Howdy!"}, 200
```

Comment (matejvasek) [reply templates/python/events/func.py:8]: also I believe that docstring for a function should be written inside `""" """` not using `#`.

Comment (matejvasek) [reply templates/python/events/func.py:8]: To be clear: I am not strong proponent of typehints, but they may be useful for APIs.

**Comment by lance** [comment]
> I discovered that this PR does not appear to work with kn func run, but works fine with docker run. I need to investigate this.

Hmm - this appears to be a problem with me locally. I'm seeing the same thing with all runtimes.

**Comment by lance** [comment]
Never mind. Seems I needed this: https://github.com/boson-project/func/commit/7ed2e86d9672f285c1def490a3d325ceb9e8471f

File: templates/python/events/func.py (hunk 1 above)
Comment (lance) [reply templates/python/events/func.py]: Done. Thanks for the suggestions.

**Pull request merged**